index.js

```diff
@@ -21,52 +28,40 @@ exports.put = function(key, value, time, timeoutCallback) {
     size++;
   }
 
-  var expire = time + now();
   var record = {
     value: value,
-    expire: expire
+    expire: time + now()
   };
 
-  if (!isNaN(expire)) {
-    var timeout = setTimeout(function() {
-      exports.del(key);
-      if (typeof timeoutCallback === 'function') {
-        timeoutCallback(key);
-      }
-    }, time);
-    record.timeout = timeout;
-  }
+  var timeout = setTimeout(function() {
+    exports.del(key);
+    if (timeoutCallback) {
+      timeoutCallback(key);
+    }
+  }, time);
+  record.timeout = timeout;
 
   cache[key] = record;
 };
 
 exports.del = function(key) {
-  var canDelete = true;
+  var deleted = false;
 
   var oldRecord = cache[key];
   if (oldRecord) {
     clearTimeout(oldRecord.timeout);
-    if (!isNaN(oldRecord.expire) && oldRecord.expire < now()) {
```

Comment:
It was added here: https://github.com/ptarjan/node-cache/commit/5e20f7d4c5c8fc2cb7c2fc297f64f97ed9b09413